File: Programmers/연습문제/Level 1/하샤드 수/Solution.js
function solution(x) {
    var answer = getIsHarshad(x);
    return answer;
}

function getIsHarshad(x) {
    const digitSum = String(x).split('').reduce((acc, add) => acc + parseInt(add));
    return Boolean(x % digitSum === 0);
}

File: Programmers/연습문제/Level 1/하샤드 수/Solution.swift
func solution(_ x:Int) -> Bool {
    return isHashad(x)
}

func isHashad(_ x: Int) -> Bool{
    let sum = getDigitSum(x)
    return (x % sum) == 0
}

func getDigitSum(_ x: Int) -> Int{
    return String(x)
        .map{ Int(String($0))! }
        .reduce(0, +)
}

// TC
let arr = 13
print(solution(arr))
